core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -208,12 +208,13 @@ class SocketServer(val config: KafkaConfig,
     val orderedAcceptors = List(dataPlaneAcceptors.get(interBrokerListener)) ++
       dataPlaneAcceptors.asScala.filterKeys(_ != interBrokerListener).values
     orderedAcceptors.foreach { acceptor =>
-      val endpoint = acceptor.endPoint
-      debug(s"Wait for authorizer to complete start up on listener ${endpoint.listener}")
-      authorizerFutures.get(endpoint).foreach { future =>
-        future.join()
+      val acceptorEndpoint = acceptor.endPoint
+      debug(s"Wait for authorizer to complete start up on listener ${acceptorEndpoint.listener}")
+      authorizerFutures.foreach { case (endpoint, future) =>
+        if (endpoint.listener().equals(acceptorEndpoint.listener()))
```

Comment:
Can we please add a comment explaining why this is neccessary? Also, `==` instead of `equals`.

Reply:
(i.e. we can't rely on `get` due to ephemeral ports, I assume).